### DISCUSSION TITLE
Are there any way to catch and save linked images?

### DISCUSSION THREAD

**Discussion by funnybunnies** [gildas-lormeau/SingleFile#852] [Q&A]
Hello.
I want to preface with apologizing/warning for me being on very casual side of user.

I got recommended this plugin to save webpage into one single file. It is good.
What I could not get to work:
For it to save linked images. 

Example: the website has thumbnails. When you click on the thumbnails it shows the bigger image in what I think is called "Lightbox" (you probably know what I mean!)

Are there any way for SingleFile to catch that and save that too? (because the thumbnails are so small for me to be of any use)

I am fully ready to accept that it is not possible, but I wanted to ask in case it is possible to do so.

**Comment by gildas-lormeau** [comment]
Unfortunately, it is not really possible to implement this feature because there is no standard way to detect and save a Lightbox component.

**Reply by thezac** [reply]
What if instead of detecting a Lightbox mechanism specifically, there is an option to save any linked images when the content of the <a> element is itself an image (or some large portion of it is an image, e.g. an image plus title text).

It looks like I can make it work with the userScript functionality, but a built in option would be very convenient for the use case described by OP.

**Reply by gildas-lormeau** [reply]
The problem is that it's impossible to know in advance if a link is pointing to an image. Thus, the only way to implement the feature would be to fetch *all* the URLs found on the page. I guess this could result in a very large number of requests.

**Comment by funnybunnies** [comment]
As I suspected then! Thank you for taking time to write!

**Comment by marco-eckstein** [comment]
The browser extension DownThemAll! ([Firefox](https://addons.mozilla.org/firefox/addon/downthemall/)/[Chrome](https://chrome.google.com/webstore/detail/downthemall/nljkibfhlpcnanjgbnlnbjecgicbjkge)) may be helpful for some websites. It does not work for Facebook galleries though.
